### PR TITLE
Add support for plugins in vimrc files

### PR DESCRIPTION
**Pull request by josephfrazier**
I referred to the Vundle documentation to implement the resolver:
https://github.com/VundleVim/Vundle.vim/blob/11fdc428fe741f4f6974624ad76ab7c2b503b73e/doc/vundle.txt#L196

Here's a file for testing:
https://github.com/gmarik/vimfiles/blob/1f4f26d42f54443f1158e0009746a56b9a28b053/vimrc#L136

Resolves https://github.com/OctoLinker/browser-extension/issues/31
Resolves https://github.com/OctoLinker/browser-extension/issues/41

### CODE REVIEW

File: lib/resolver/vim-plugin.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
+// Logic adapted from https://github.com/VundleVim/Vundle.vim/blob/11fdc428fe741f4f6974624ad76ab7c2b503b73e/doc/vundle.txt#L196
+export default function ({ shorthand }) {
+  const components = shorthand.split('/');
+
+  // GitHub is used when a user/repo is passed to `Plugin`.
+  if (components.length === 2) {
+    return ['https://github.com/' + shorthand];
```
Comment (stefanbuck) [inline lib/resolver/vim-plugin.js]: Please use the shorthand package.

```js
import ghShorthand from 'github-url-from-username-repo';
ghShorthand(shorthand, true);
```

File: lib/resolver/vim-plugin.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  }
+  // Any single word without a slash '/' is assumed to be from Vim Scripts.
+  if (components.length === 1) {
+    return ['https://github.com/vim-scripts/' + shorthand];
+  }
+  // Assume it's a URL otherwise. We can't link to git/ssh, so change to https
+  // and hope it works.
+  return [
+    shorthand.replace(/^git|ssh/, 'https'),
+    shorthand.replace(/^git|ssh/, 'http'),
```
Comment (stefanbuck) [inline lib/resolver/vim-plugin.js]: I would prefer to use a package for that https://www.npmjs.com/package/giturl

File: lib/resolver/vim-plugin.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+export default function ({ shorthand }) {
+  const components = shorthand.split('/');
+
+  // GitHub is used when a user/repo is passed to `Plugin`.
+  if (components.length === 2) {
+    return ['https://github.com/' + shorthand];
+  }
+  // Any single word without a slash '/' is assumed to be from Vim Scripts.
+  if (components.length === 1) {
+    return ['https://github.com/vim-scripts/' + shorthand];
```
Comment (stefanbuck) [inline lib/resolver/vim-plugin.js]: This doesn't have to be an array. A resolver can return either an Array of urls or just a url string.

**Comment by stefanbuck** [comment]
Amazing, besides a few comments this looks really good 👍

File: lib/resolver/vim-plugin.js [hunk 4]
```diff
@@ -1,14 +1,16 @@
...
 
   // GitHub is used when a user/repo is passed to `Plugin`.
   if (components.length === 2) {
-    return ['https://github.com/' + shorthand];
+    return [ghShorthand(shorthand, true)];
   }
   // Any single word without a slash '/' is assumed to be from Vim Scripts.
   if (components.length === 1) {
-    return ['https://github.com/vim-scripts/' + shorthand];
+    return [ghShorthand(`vim-scripts/${shorthand}`, true)];
```
Comment (stefanbuck) [inline lib/resolver/vim-plugin.js]: Nice 👍

**Comment by stefanbuck** [comment]
Thanks @josephfrazier

**Comment by josephfrazier** [comment]
Thanks for the quick feedback/merge!

**Pull request merged**